Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #69840
* #68462
* #68693
* #68692
* #68691
* #68697
* #68690
* #68689
* #68714
* __->__ #68688
* #68687
* #68247

This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

peterbell10 committed:
Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro

This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

[ghstack-poisoned]

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

[ghstack-poisoned]

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

[ghstack-poisoned]

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

[ghstack-poisoned]

albanD commented on aten/src/ATen/templates/DispatchKeyFunctions_inl.h:
```diff
@@ -7,6 +7,14 @@
 #include <c10/core/Scalar.h>
 #include <ATen/core/Reduction.h>
 
+// TODO: If necessary, consider adding <ATen/ops/{function}_key.h> headers
```
Is this for you or future devs?

Do we want to write a "note" somewhere about the function specific headers and have all the error message point to it?

peterbell10 replied:
Sorry, just noticed my comment didn't get posted. This was a note to myself, referring to splitting this header into per-operator headers. The next PR in the stack resolves that and the comment is gone.

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32596267).

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32596267).

peterbell10 committed:
Rebase to fix merge conflicts on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32596267).

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

peterbell10 committed:
Fix mypy issues on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

peterbell10 committed:
Commit to re-run CI on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

peterbell10 committed:
Update on "Add TORCH_ASSERT_ONLY_METHOD_OPERATORS macro"


This adds a new macro `TORCH_ASSERT_ONLY_METHOD_OPERATORS` which
allows `Tensor.h` to be included, but not headers which pull in all
other operators. So, a file that defines this macro needs to use the
fine-grained headers to include only the operators being used.

Differential Revision: [D32596267](https://our.internmc.facebook.com/intern/diff/D32596267)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32596267).
